src/algebra/algebra/basic.lean

```diff
@@ -183,6 +183,18 @@ end
 
 end field_nontrivial
 
+lemma coe_inv {R A : Type*}
+  [field R] [division_ring A] [algebra R A] (x : R) :
+  (↑x : A)⁻¹ = ↑x⁻¹ :=
+begin
+  rcases ne_or_eq x 0 with hn | rfl,
+  { apply inv_eq_of_mul_eq_one_left,
+    rwa [←algebra_map.coe_mul, inv_mul_cancel, algebra_map.coe_one] },
+  { push_cast,
+    simp_rw inv_zero,
+    rw algebra_map.coe_zero }
+end
```

Comment:
This should just be `algebra_map.map_inv\0` or similar

Reply:
And you should add the similar lemmas for `div` and `zpow`

Reply:
> This should just be algebra_map.map_inv\0 or similar

I don't understand; there are existing lemmas `algebra_map.coe_mul`, `coe_one`, etc, but not one for `inv`; so I made one and added it to the same file as the existing lemmas, that's what this is. Are you saying they should be renamed from `coe_xxx` to `map_xxx`?

Reply:
Sorry, what I mean is that you should be able to prove this statement much more quickly with `(algebra_map R A).map_inv\0 x` .

Reply:
Ok, I've rewritten it with `map_inv0`, which avoids needing to edit `algebra.algebra.basic`. Waiting for CI now.

Reply:
Oh, I meant that the lemma should exist as you stated it, but the proof should be that one line. I can add it in a separate PR if that's easier for you.

Reply:
Thanks, I'd rather keep it separate from this PR -- if I edit a low-level file like `algebra.algebra.basic`, and then try to do anything in `special_functions` at the other end of the import hierarchy, I then have to wait half an hour for 1000 intermediate files to recompile!

Reply:
#18422